buildspecs/buildspec-rust.yml

```diff
@@ -0,0 +1,18 @@
+version: 0.2
+
+phases:
+  install:
+    commands:
+      - pip install --upgrade pip aws-sam-cli
+      - pip install -r requirements.txt
+      # Install rustup
+      - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+      # Install cross for cross-compilation
+      - cargo install cross
+  
+  post_build:
+    commands:
+      - sam --version
+      # sam build doesn't support building Rust Lambda functions
```

Comment:
Can we use custom build option for rust?
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/building-custom-runtimes.html

Reply:
Yes, but not without serious drawbacks. `sam build` with Makefile copies all the files in the directory to a temporary folder.

In Rust, you'll typically setup cargo at the root of your repository (and not in a sub-folder for a specific Lambda function), and you can use a single crate or workspace to build multiple functions. That helps minimize the amount of recompilation. When doing unit tests, you compile and store a lot of artefacts into the `target/` folder. That means that that folder could be 5-20 GB in size.

That meant that building 4 Lambda functions took ~26 minutes with `sam build` versus less than 30 seconds without it. The majority of the time was spent copying files.

Reply:
(marking as resolved per our discussion)